xblock/plugin.py

```diff
@@ -41,6 +45,37 @@ def default_select(identifier, all_entry_points):  # pylint: disable=inconsisten
         raise AmbiguousPluginError(all_entry_points)
 
 
+def default_select(identifier, all_entry_points):
+    """
+    Selects plugin for the given identifier with the ability for a Plugin to override
+    the default entry point.
+
+    Raises:
+    - PluginMissingError when we don't have an entry point or entry point to override.
+    - AmbiguousPluginError when we have ambiguous entry points.
+    """
+
+    # Split entry points into overrides and non-overrides
+    overrides = []
+    block_entry_points = []
+
+    for block_entry_point in all_entry_points:
+        if block_entry_point.group.endswith('.overrides'):
+            overrides.append(block_entry_point)
+        else:
+            block_entry_points.append(block_entry_point)
+
+    # Get the default entry point
+    default_plugin = _default_select_no_override(identifier, block_entry_points)
+
+    # If we have an unambiguous override, that gets priority. Otherwise, return default.
+    if len(overrides) == 1:
+        return overrides[0]
+    elif len(overrides) > 1:
+        raise AmbiguousPluginError(all_entry_points)
+    return default_plugin
```

Comment:
Regarding _default_select_no_override -- Yeah, I do not feel strongly since it is an internal implementation detail, but combining the functions does make it easier to follow IMO. If you are willing to make that change @nsprenkle, great; if not, I don't want to block merging on this concern.

Regarding the error message -- @farhan makes a very good point, we should make it clear whether there are multiple regular plugins vs. multiple overrides. As it is, the error cases would be indistinguishable to an operator. 

@nsprenkle , I don't have a preference between a new exception class vs. the same exception with a different message. If you do go with separate exceptions, please just make AmbiguousPluginOverrideError a subclass of AmbiguousPluginError so that existing `catch AmbiguousPluginError` statements will catch AmbiguousPluginOverrideError.

Reply:
@kdmccormick , added as a new Exception